app/assets/javascripts/discourse/app/services/group-automatic-members-dialog.js

```diff
@@ -8,12 +8,12 @@ export default class GroupAutomaticMembersDialog extends Service {
   @service dialog;
 
   async showConfirm(group_id, email_domains) {
-    const domainCount = email_domains?.split("|")?.length ?? 0;
```

Comment:
The problem was here. In JavaScript:

```js
"".split("|")
// ['']
```

making it think there is one domain when there is in fact none.